### DISCUSSION TITLE
APatch Support

### DISCUSSION THREAD

**Discussion by DerGoogler** [MMRLApp/MMRL#106] [Announcements]
> [!IMPORTANT]
> Due some complaints within the Android Community LSPosed, KernelSU and many others has been achieved, unfortunately.. I'll try to keep the KernelSU support upright!

I need some information to add possible APatch support:

- where is the `su` binary located
  - `which su`
- where and how are the installed modules stored? Same as Magisk and KernelSU?
  - `/data/adb`?
- does there exist a module install command line tool?
  - something like `magisk --install-module <ZIP>` or `ksud module install <ZIP>`
- how to detect APatch?
  - `su -c cat /proc/self/mounts > /sdcard/ap_mounts.txt` and send me them on Telegram ([@Der_Googler](https://t.me/Der_Googler))

**Comment by RashadGasimly** [comment]
1."su" is located on /data/adb/apd
2.yes
3.idk
4.idk
![IMG_20240111_232614_652.jpg](https://github.com/DerGoogler/MMRL/assets/129990317/5422d1e6-e001-4455-8ccb-e270648ef7d6)

Here is a photo of apatch

**Comment by DerGoogler** [comment]
@RashadGasimly

Thanks for the Infos, very helpful for the future development. Could you execute `su -h` or `su --help` so I can see what you can to with it. It would also be awesome if you can send the folder structure of `/data/adb/apd`

Use the Termux `tree` command for this:
```shell
/data/data/com.termux/files/usr/bin/tree /data/adb/apd
```

> You'll probably need install this command by `pkg install tree`

**Comment by DerGoogler** [comment]
APatch will coming to MMRL
